public/widgets/widget.js

```diff
@@ -0,0 +1,355 @@
+   * @type {string}
+   */
+  var loaderText = 'loading widget';
+
+  /**
+   * Element for the loading animation
+   * @type {Element}
+   */
+  var loaderElement;
+
+  /**
+   * Contains the interval id for the animation loop
+   * @type {Number}
+   */
+  var loaderInterval;
+
+  /**
+   * Checks the settings given to the function and starts
+   * the widget creation.
+   */
+  var construct = function () {
+    // fix for #135, all but the first forward slash of a DOI
+    // should be encoded as %2F
+    if (settings.hasOwnProperty('DOI')) {
+      var tmp = settings.DOI.split('/');
+      settings.DOI = tmp[0] + '/' + tmp.slice(1).join('%2F');
+    }
+
+    containerElement = document.getElementsByClassName(settings.containerClass)[0];
+
+    // Changes the ORCID widget type to DOI if the user requests it.
+    if (settings.hasOwnProperty('ORCIDWidgetType') && settings.ORCIDWidgetType === 'DOI') {
+      ORCIDWidgetType = 'DOI';
+    }
+
+    if (settings.hasOwnProperty('loaderText')) {
+      loaderText = settings.loaderText;
+    }
+
+    // we need a container element and either a DOI or an ORCID
+    // proceed if these are available, throw an error if not
+    if (containerElement && (settings.hasOwnProperty('DOI') || settings.hasOwnProperty('ORCID'))) {
+      insertCSS();
+      startLoader();
+      loadWidgetContent();
+    } else {
+      throw 'The settings object is incomplete. You need to supply an element id and either a DOI or an ORCID.';
+    }
+  };
+
+  /**
+   * Injects a style tag into the header containing the necessary
+   * styles for the widget.
+   */
+  var insertCSS = function () {
+    var css = '.paper-badge {float: left; width: 10em; height: 20em; overflow: hidden; ' +
+      'border-top: 1px solid #ccc; height 15em; padding: 2%; margin-right: 1%; margin-top: 2%}' +
+      '.paper-badge a {width: 100%; display: inline-block; font-size: 88%; line-height: 1.2; ' +
+      'color: #333; padding: 0.4em; cursor: pointer; text-decoration: none} .paper-badge ' +
+      'a.active {color: #fff; background: #7ab441; text-decoration: none} .paper-badge img ' +
+      '{max-width: 8em; margin-left: 10%; margin-bottom: 1%} .paper-badges-hidden{display: none}';
+
+    var head = document.head || document.getElementsByTagName('head')[0];
+    var style = document.createElement('style');
+    style.type = 'text/css';
+    if (style.styleSheet) {
+      style.styleSheet.cssText = css;
+    } else {
+      style.appendChild(document.createTextNode(css));
+    }
+
+    head.appendChild(style);
+  };
+
+  /**
+   * Inserts the loader element into the container element
+   * and starts an animation loop.
+   */
+  var startLoader = function () {
+    loaderElement = document.createElement('div');
+    loaderElement.className = 'paperbadger-loader';
+    loaderElement.setAttribute('data-step', '4');
+    containerElement.appendChild(loaderElement);
+    loaderMethod();
+
+    loaderInterval = setInterval(loaderMethod, 750);
```

Comment:
Thanks! I first thought I add a base64 encoded gif to the source, but then thought about file size and performance, so I did it that way.